docs/router.md

```diff
@@ -0,0 +1,300 @@
+
+If a `gpii.express.router` instance is given the chance to handle a request, it evaluates its "stack" in order,
+to find a piece of middleware whose path, [HTTP method](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html), and
+signature all match the current `request`.  See the [middleware](middleware.md) docs for details on the method signature
+and HTTP method.  The effective path for a given router module is determined by appending its path to that of its
+immediate parent.  A `path` of `/` indicates that a router accepts all requests that its parent does.
+
+If a match is found, the middleware is given the chance to work with a given request.  If a match is not found, the
+next piece of middleware in the "stack" is evaluated.  If the end of the "stack" is reached without finding any matching
+middleware, the parent container (either `gpii.express` or another `gpii.express.router` instance) will continue
+evaluating its own "stack".  If the "stack of stacks" has been exhausted and no middleware is found to match a given
+request method and path, Express itself will report a 404 error.
+
+# Ordering Middleware and Routers
+
+When you are working only with `gpii.express` itself and `gpii.express.middleware`, there is only one "stack", that of
+`gpii.express` itself, which is in effect a "chain" of middleware in order.  When you start working with one or more
+`gpii.express.router` instances, you have a "stack of stacks", or something that is more of a "tree".  The "tree" is
+[traversed in "pre-order" fashion](https://en.wikipedia.org/wiki/Tree_traversal#Pre-order), as illustrated here:
+
+![A diagram demonstrating the order in which routing occurs within `gpii.express`.]( https://upload.wikimedia.org/wikipedia/commons/d/d4/Sorted_binary_tree_preorder.svg "Tree traversal diagram.")
+
+This image and the explanation are taken from the [Wikipedia article on "tree traversal"](https://en.wikipedia.org/wiki/Tree_traversal).
+The order in which these elements are given the chance to work with a request is F, B, A, D, C, E, G, I, and finally H.
+You can control the order in which middleware is added to its enclosing "stack" using
+[namespaces and priorities](http://docs.fluidproject.org/infusion/development/Priorities.html).
+
+The above diagram can be modeled using the following component options:
+
+```
+fluid.defaults("my.enclosing.component", {
+    gradeNames: ["fluid.component"],
+    components: {
+        f: {
+            type: "gpii.express",
+            options: {
+                port: 9090,
+                components: {
+                    b: {
+                        type: "gpii.express.router",
+                        options: {
+                            priority: "before:g",
+                            components: {
+                                a: {
+                                    type: "my.namespaced.middleware"
+                                },
+                                b: {
```

Comment:
Is this component name `b` ambiguous with the same `b` component at line 48? Is this intentional? Why?